katana/katana.py

```diff
@@ -90,6 +93,227 @@ def readconfig():
+    tree = xml.etree.ElementTree.fromstring(corrected_xml)
+    return tree
+
+
+@route('/parsexmlobj', method='POST')
+def parsexmlobj():
+    """
+    This method fetch the XML object from the Katana UI of Keyword sequencing tool screen.
+    And parse it to form a new Wrapper keyword & place it in the user provided file path.
+    """
+    method_names = []
+    sample = []
+    xmlobj = parseString("".join(request.body))
+    tree = get_correct_xml_and_root_element(xmlobj)
+
+    WrapperName = tree[0][0].text
+    # To get the wrapper keyword name, Action file name and the
+    # description which is provided by the user.
+    ActionFile = tree[0][2].text.strip()
+    Description = tree[0][3].text
+
+    with open(ActionFile, 'r') as files:
+        # To get the method names available in user provided
+        # action file from keyword sequencing tool UI screen
+        for line in files:
+            next_word = get_method_name(line)
+            method_names.append(next_word)
+
+    if WrapperName in method_names:
+        # To check if the user provided wrapper keyword name already
+        # exists in action file
+        checkval = "yes"
+    else:
+        checkval = "no"
+        Subkeyword = tree.find('Subkeyword')
+        subkw_list = Subkeyword.findall('Skw')
+        doc_string_value, subkw_list, object_list_new, import_action_list, kw_name_1 = check_action_file_name(subkw_list, ActionFile)
+        sum_val = 1
+        for new in doc_string_value:
+            # To have the doc string in the order with
+            # numbers
+            sample.append('\n' + "        " + str(sum_val) + ". " + new)
+            sample_string = "".join(sample)
+            sum_val += 1
+            if sum_val > len(subkw_list):
+                break
+
+        template_dir, template_dir_1 = get_relative_path_for_kw_seq_temp()
+
+        for line in io.open(template_dir, 'r'):
+            # Replacing kw_seq_template with the respective
+            # details and placing it in user provided action file
+            with open(ActionFile, 'a+') as f2:
+                line = line.replace('$new_line', '\n')
+                line = line.replace('$wrapper_kw', WrapperName)
+                line = line.replace('$doc_string_values', sample_string)
+                if Description is not None:
+                    line = line.replace('$wdesc', Description)
+                else:
+                    line = line.replace('$wdesc', 'Description not provided by the user')
+                line = line.replace('$kw_name_1', str(kw_name_1))
+                f2.write(line)
+        final_obj_list = set(object_list_new)
+        final_import_list = set(import_action_list)
+        with open(ActionFile, 'a+') as f2:
+            for _, vals in enumerate(final_import_list):
+                vals = vals.decode('utf-8')
+                f2.write('\n' + "        " + vals)
+            for _, val in enumerate(final_obj_list):
```

Comment:
enumerate not required in line 179 & 182

Reply:
done